doc/src/manual/environment-variables.md

```diff
@@ -0,0 +1,138 @@
+```
+/home/alice/.julia/v0.5
+```
+
+### `JULIA_HISTORY`
+The absolute path [`Base.find_hist_file()`](https://github.com/JuliaLang/julia/blob/master/base/REPL.jl#L615) of the REPL's history file. If `JULIA_HISTORY` is not specified, then `Base.find_hist_file()` defaults to
+```
+$HOME/.julia_history
+```
+
+### `JULIA_PKGRESOLVE_ACCURACY`
+A positive `Int` that determines how much time the max-sum subroutine [`MaxSum.maxsum()`](https://github.com/JuliaLang/julia/blob/master/base/pkg/resolve/maxsum.jl#L442) of the package dependency resolver [`Resolve.resolve()`](https://github.com/JuliaLang/julia/blob/master/base/pkg/resolve.jl#L15)  will devote to attempting satisfying constraints before giving up: this value is by default `1`, and larger values correspond to larger amounts of time.
+
+Suppose the value of `JULIA_PKGRESOLVE_ACCURACY` is `n`. Then
+
+*   the number of pre-decimation iterations is `20*n`,
+*   the number of iterations between decimation steps is `10*n`, and
+*   at decimation steps, at most one in every `20*n` packages is decimated. 
+
+## External applications
+
+### `JULIA_SHELL`
+The absolute path of the shell with which Julia should execute external commands (via [`Base.repl_cmd()`](https://github.com/JuliaLang/julia/blob/master/base/client.jl#L91)). Defaults to the environment variable `$SHELL`, and falls back to `/bin/sh` if `$SHELL` is unset. 
+
+*Note:* the `fish` shell is unsupported.
+
+### `JULIA_EDITOR`, `VISUAL`, `EDITOR`
+The editor returned by [`Base.editor()`](https://github.com/JuliaLang/julia/blob/master/base/interactiveutil.jl#L12) and used in, e.g., [`Base.edit()`](https://github.com/JuliaLang/julia/blob/master/base/interactiveutil.jl#L33). `JULIA_EDITOR` takes precedence over `VISUAL`, which in turn takes precedence over `EDITOR`. If none of these environment variables is set, then the editor is taken to be `open` on Windows and OS X, or `/etc/alternatives/editor` if it exists, or `emacs` otherwise.
+
+**Warning:** `JULIA_EDITOR` is *not* used in the determination of the editor for [`Pkg.edit()`](https://github.com/JuliaLang/julia/blob/master/base/pkg/entry.jl#L36): this function checks `VISUAL` and `EDITOR` alone.
+
+## Parallelization
+
+### `JULIA_CPU_CORES`
+Overrides the global variable [`Sys.CPU_CORES`](https://github.com/JuliaLang/julia/blob/master/base/sysinfo.jl#L62), the number of logical CPU cores available. 
+
+### `JULIA_WORKER_TIMEOUT`
+A `Float64` that sets the value of [`Base.worker_timeout()`](https://github.com/JuliaLang/julia/blob/master/base/multi.jl#L1566) (default: `60.0`). This function gives the number of a seconds a worker process will wait for a master process to establish a connection before dying.
+
+## REPL formatting
+Environment variables that determine how REPL output should be formatted at the terminal. Generally, these variables should be set to [ANSI terminal escape sequences](http://ascii-table.com/ansi-escape-sequences.php).
+
+###   `JULIA_ERROR_COLOR`
```

Comment:
These have 3 spaces after the hashes. Was that intentional? For consistency it would be better to use a single space.